File: SretneSapice/UI/sretnesapice_mobile/lib/models/dog_walker.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sretnesapice_mobile/models/city.dart';
import 'package:sretnesapice_mobile/models/service_request.dart';
import 'package:sretnesapice_mobile/models/walker_review.dart';

part 'dog_walker.g.dart';

@JsonSerializable()
class DogWalker {
  int? dogWalkerId;
  int? userId;
  String? name;
  String? surname;
  String? fullName;
  int? age;
  int? cityId;
  String? phone;
  String? experience;
  String? dogWalkerPhoto;
  int? rating;
  bool? isApproved;
  String? status;
  City? city;
  List<WalkerReview> walkerReviews;
  List<ServiceRequest> serviceRequests;

  DogWalker(
      this.dogWalkerId,
      this.userId,
      this.name,
      this.surname,
      this.fullName,
      this.age,
      this.cityId,
      this.phone,
      this.experience,
      this.dogWalkerPhoto,
      this.rating,
      this.isApproved,
      this.status,
      this.city,
      this.walkerReviews,
      this.serviceRequests);

  factory DogWalker.fromJson(Map<String, dynamic> json) =>
      _$DogWalkerFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$DogWalkerToJson(this);
}

File: SretneSapice/SretneSapice.Services/Database/DogWalker.cs
using System;
using System.Collections.Generic;

namespace SretneSapice.Services.Database;

public partial class DogWalker
{
    public int DogWalkerId { get; set; }

    public int? UserId { get; set; }

    public int? CityId { get; set; }

    public string Name { get; set; } = null!;

    public string Surname { get; set; } = null!;

    public int Age { get; set; }

    public string Phone { get; set; } = null!;

    public string Experience { get; set; } = null!;

    public byte[] DogWalkerPhoto { get; set; } = null!;
    public byte[]? DogWalkerPhotoThumb { get; set; } = null!;

    public int? Rating { get; set; }

    public bool? IsApproved { get; set; }

    public string Status { get; set; }

    public virtual City? City { get; set; }

    public virtual ICollection<DogWalkerAvailability> DogWalkerAvailabilities { get; } = new List<DogWalkerAvailability>();

    public virtual ICollection<DogWalkerLocation> DogWalkerLocations { get; } = new List<DogWalkerLocation>();

    public virtual ICollection<FavoriteWalker> FavoriteWalkers { get; } = new List<FavoriteWalker>();

    public virtual ICollection<ServiceRequest> ServiceRequests { get; } = new List<ServiceRequest>();

    public virtual User? User { get; set; }

    public virtual ICollection<WalkerReview> WalkerReviews { get; } = new List<WalkerReview>();
}
